src/buy/components/BuyDropdown.tsx

```diff
@@ -0,0 +1,110 @@
+import { useOnchainKit } from '@/core-react/useOnchainKit';
+import { openPopup } from '@/ui-react/internal/utils/openPopup';
+import { useCallback, useEffect, useMemo } from 'react';
+import { useAccount } from 'wagmi';
+import { getRoundedAmount } from '../../core/utils/getRoundedAmount';
+import { ONRAMP_BUY_URL } from '../../fund/constants';
+import { getFundingPopupSize } from '../../fund/utils/getFundingPopupSize';
+import { background, cn, color, text } from '../../styles/theme';
+import { ONRAMP_PAYMENT_METHODS } from '../constants';
+import { BuyOnrampItem } from './BuyOnrampItem';
+import { useBuyContext } from './BuyProvider';
+import { BuyTokenItem } from './BuyTokenItem';
+
+export function BuyDropdown() {
+  const { projectId } = useOnchainKit();
+  const { to, fromETH, fromUSDC, from, startPopupMonitor, setIsDropdownOpen } =
+    useBuyContext();
+  const { address } = useAccount();
+
+  const handleOnrampClick = useCallback(
+    (paymentMethodId: string) => {
+      return () => {
+        const assetSymbol = to?.token?.symbol;
+        let fundAmount = to?.amount;
+        // funding url requires a leading zero if the amount is less than 1
+        if (fundAmount?.[0] === '.') {
+          fundAmount = `0${fundAmount}`;
+        }
+        const fundingUrl = `${ONRAMP_BUY_URL}/one-click?appId=${projectId}&addresses={"${address}":["base"]}&assets=["${assetSymbol}"]&presetCryptoAmount=${fundAmount}&defaultPaymentMethod=${paymentMethodId}`;
+        const { height, width } = getFundingPopupSize('md', fundingUrl);
+        const popupWindow = openPopup({
+          url: fundingUrl,
+          height,
+          width,
+          target: '_blank',
+        });
+
+        if (popupWindow) {
+          // Detects when the popup is closed
+          // to stop loading state
+          startPopupMonitor(popupWindow);
+        }
+      };
+    },
+    [address, to, projectId, startPopupMonitor],
+  );
+
+  const formattedAmountUSD = useMemo(() => {
+    if (!to?.amountUSD || to?.amountUSD === '0') {
+      return null;
+    }
+    const roundedAmount = Number(getRoundedAmount(to?.amountUSD, 2));
+    return `$${roundedAmount.toFixed(2)}`;
+  }, [to?.amountUSD]);
```

Comment:
Doesn't have to be in this PR but flagging that we should have a utility function for this. 

`SwapAmountInput` needs the same:

https://github.com/coinbase/onchainkit/blob/paul/docs-remove-old/src/swap/components/SwapAmountInput.tsx#L74-L80